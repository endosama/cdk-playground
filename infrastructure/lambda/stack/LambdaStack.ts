import * as cdk from "@aws-cdk/core";
import * as lambda from "@aws-cdk/aws-lambda";

export class LambdaStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id);

    new lambda.Function(this, "MyLambda", {
      runtime: lambda.Runtime.NODEJS_12_X,
      code: lambda.Code.fromAsset(`${"./src/lambda/"}`),
      handler: "BaseLambda.handler",
      tracing: lambda.Tracing.ACTIVE,
    });
  }
}
