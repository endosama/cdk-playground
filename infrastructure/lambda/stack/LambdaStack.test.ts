import { LambdaStack } from "./LambdaStack";
import * as cdk from "@aws-cdk/core";
import { SynthUtils } from "@aws-cdk/assert";
import "@aws-cdk/assert/jest";

const mockedEnv = {
  env: {
    account: "111111111111",
    region: "eu-west-2",
  },
};

describe("LambdaStack", () => {
  it("should match the snapshot", () => {
    const app = new cdk.App();
    const stack = new LambdaStack(app, "Test", mockedEnv);
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });

  it("should contain a lambda function with the correct name", () => {
    const app = new cdk.App();
    const stack = new LambdaStack(app, "Test", mockedEnv);
    expect(stack).toHaveResource("AWS::Lambda::Function");
  });
});
