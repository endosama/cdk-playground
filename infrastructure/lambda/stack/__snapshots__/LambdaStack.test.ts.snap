// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LambdaStack should match the snapshot 1`] = `
Object {
  "Parameters": Object {
    "AssetParametersebc85ccfa99c37e95c2081f1bae88d02fc47fa5efa3a88f41ccfad0a23880bbcArtifactHashDB0503D2": Object {
      "Description": "Artifact hash for asset \\"ebc85ccfa99c37e95c2081f1bae88d02fc47fa5efa3a88f41ccfad0a23880bbc\\"",
      "Type": "String",
    },
    "AssetParametersebc85ccfa99c37e95c2081f1bae88d02fc47fa5efa3a88f41ccfad0a23880bbcS3BucketF62D5450": Object {
      "Description": "S3 bucket for asset \\"ebc85ccfa99c37e95c2081f1bae88d02fc47fa5efa3a88f41ccfad0a23880bbc\\"",
      "Type": "String",
    },
    "AssetParametersebc85ccfa99c37e95c2081f1bae88d02fc47fa5efa3a88f41ccfad0a23880bbcS3VersionKey318B5161": Object {
      "Description": "S3 key for asset version \\"ebc85ccfa99c37e95c2081f1bae88d02fc47fa5efa3a88f41ccfad0a23880bbc\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "MyLambdaCCE802FB": Object {
      "DependsOn": Array [
        "MyLambdaServiceRoleDefaultPolicy5BBC6F68",
        "MyLambdaServiceRole4539ECB6",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersebc85ccfa99c37e95c2081f1bae88d02fc47fa5efa3a88f41ccfad0a23880bbcS3BucketF62D5450",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersebc85ccfa99c37e95c2081f1bae88d02fc47fa5efa3a88f41ccfad0a23880bbcS3VersionKey318B5161",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersebc85ccfa99c37e95c2081f1bae88d02fc47fa5efa3a88f41ccfad0a23880bbcS3VersionKey318B5161",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "BaseLambda.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "MyLambdaServiceRole4539ECB6",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "TracingConfig": Object {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "MyLambdaServiceRole4539ECB6": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "MyLambdaServiceRoleDefaultPolicy5BBC6F68": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "MyLambdaServiceRoleDefaultPolicy5BBC6F68",
        "Roles": Array [
          Object {
            "Ref": "MyLambdaServiceRole4539ECB6",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
